sql/analyzer/aliases.go

```diff
@@ -306,6 +306,9 @@ func disambiguateTableFunctions(ctx *sql.Context, a *Analyzer, n sql.Node, scope
 	return transform.Node(n, func(n sql.Node) (sql.Node, transform.TreeIdentity, error) {
 		switch n := n.(type) {
 		case *expression.UnresolvedTableFunction:
+			if n.Alias != "" {
```

Comment:
probably OK to wrap the constructors in `parse.go` `planbuilder`, delete the logic here